database/connection.js

```diff
@@ -12,50 +12,94 @@ const bookshelf = require('bookshelf')(knex);
   tableName: 'STATION',
-  trains: () => this.hasMany(Train, ['TRAIN.id']).through(StationAndTrain),
-  passengers: () => this.hasMany(Passenger, ['PASSENGER.id']).through(StationAndPassenger),
-  currentTrain: () => this.hasMany(Train, ['LastStation']),
-  tickets: ()=> this.hasMany(Ticket, ['STATION.id'])
+  trains: () => this.hasMany(Train).through(StationAndTrain),
+  passengers: () => this.hasMany(Passenger).through(StationAndPassenger),
+  currentTrain: () => this.hasMany(Train),
+  tickets: ()=> this.hasMany(Ticket)
 })
 
+var Stations = bookshelf.Collection.extend({
+  model: Station
+});
+
 const Passenger = bookshelf.Model.extend({
   tableName: 'PASSENGER',
-  trains: () => this.hasMany(Train, ['TRAIN.id']).through(TrainAndPassenger),
-  stations: () => this.hasMany(Station, ['STATION.id']).through(StationAndPassenger),
-  tickets: () => this.hasMany(Ticket, ['PASSENGER.id'])
+  trains: () => this.hasMany(Train).through(TrainAndPassenger),
+  stations: () => this.hasMany(Station).through(StationAndPassenger),
+  tickets: () => this.hasMany(Ticket)
 })
 
+var Passengers = bookshelf.Collection.extend({
+  model: Passenger
+});
+
 const Ticket = bookshelf.Model.extend({
   tableName: 'TICKET',
-  passenger: () => this.belongsTo(Passenger, ['PASSENGER.id']),
-  station: () => this.belongsTo(Station, ['STATION.id'])
+  passenger: () => this.belongsTo(Passenger),
+  station: () => this.belongsTo(Station)
 })
 
+var Tickets = bookshelf.Collection.extend({
+  model: Ticket
+});
+
 const StationAndTrain = bookshelf.Model.extend({
   tableName: 'STATION_AND_TRAIN',
-  station: () => this.belongsTo(Station, ['STATION.id']),
-  train: () => this.belongsTo(train, ['TRAIN.id'])
+  idAttribute: null,
+  station: () => this.belongsTo(Station),
+  train: () => this.belongsTo(Train)
 })
 
+var StationAndTrainRecords = bookshelf.Collection.extend({
+  model: StationAndTrain
+});
+
 const StationAndPassenger = bookshelf.Model.extend({
   tableName: 'STATION_AND_PASSENGER',
-  station: () => this.belongsTo(Station, ['STATION.id']),
-  passenger: () => this.belongsTo(Passenger, ['PASSENGER.id'])
+  idAttribute: null,
+  station: () => this.belongsTo(Station),
+  passenger: () => this.belongsTo(Passenger)
+})
+
+const StationAndPassengerRecords = bookshelf.Collection.extend({
+  model: StationAndPassenger
 })
 
 const TrainAndPassenger = bookshelf.Model.extend({
   tableName: 'TRAIN_AND_PASSENGER',
-  train: () => this.belongsTo(Train, ['TRAIN.id']),
-  passenger: () => this.belongsTo(Passenger, ['PASSENGER.id'])
+  idAttribute: null,
+  train: () => this.belongsTo(Train),
+  passenger: () => this.belongsTo(Passenger)
 })
 
-console.log("COMPLETE")
+const TrainAndPassengerRecords = bookshelf.Collection.extend({
+  model: StationAndPassenger
+})
 
-module.exports = bookshelf
+module.exports = {
+  bookshelf,
+  Train,
+  Trains,
+  Passenger,
+  Passengers,
+  Station,
+  Stations,
+  Ticket,
+  Tickets,
+  StationAndTrain,
```

Comment:
I'd probably rename your join tables to
station_trains
station_train_records
station_passenger_records
train_passengers
train_passenger_records

these are more standard based on my knowlege